### PR TITLE
MOSIP-35094 MOSIP-35110 MOSIP-35065 35096 Added parameter true to idrepo core and identity service

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

**Review by dhanendra06 — APPROVED** [review]

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoDraftServiceImpl.java [hunk 1]
```diff
@@ -274,6 +274,7 @@ private void updateDemographicData(IdRequestDTO request, UinDraft draftToUpdate)
 			Configuration configuration = Configuration.builder().jsonProvider(new JacksonJsonProvider())
 					.mappingProvider(new JacksonMappingProvider()).build();
 			DocumentContext inputData = JsonPath.using(configuration).parse(requestDTO.getIdentity());
+			((LinkedHashMap) ((JsonContext) inputData).json()).put("UIN", null);
```
Comment (dhanendra06) [inline id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoDraftServiceImpl.java:277]: why you are putting UIN as null for inputData

File: id-repository/id-repository-core/src/main/java/io/mosip/idrepository/core/exception/IdRepoExceptionHandler.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ protected ResponseEntity<Object> handleNoResourceFoundException(NoResourceFoundE
 				"handleNoResourceFoundException - \n" + ExceptionUtils.getStackTrace(Objects.isNull(rootCause) ? noResourceFoundException : rootCause));
 
 		IdRepoAppException idRepoAppException = new IdRepoAppException(INVALID_REQUEST.getErrorCode(), INVALID_REQUEST.getErrorMessage());
-			return new ResponseEntity<>(buildExceptionResponse(idRepoAppException, ((ServletWebRequest)request).getHttpMethod(), null), HttpStatus.OK);
+			return new ResponseEntity<>(buildExceptionResponse(idRepoAppException, ((ServletWebRequest)request).getHttpMethod(), null), HttpStatus.NOT_FOUND);
```
Comment (dhanendra06) [inline id-repository/id-repository-core/src/main/java/io/mosip/idrepository/core/exception/IdRepoExceptionHandler.java:108]: Httpstatus should be 200 for all the api response.

**Review by dhanendra06 — CHANGES_REQUESTED** [review]

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoDraftServiceImpl.java (hunk 1 above)
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoDraftServiceImpl.java]: removed

File: id-repository/id-repository-core/src/main/java/io/mosip/idrepository/core/exception/IdRepoExceptionHandler.java (hunk 2 above)
Comment (kameshsr) [reply id-repository/id-repository-core/src/main/java/io/mosip/idrepository/core/exception/IdRepoExceptionHandler.java]: corrected

**Review by dhanendra06 — APPROVED** [review]

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**